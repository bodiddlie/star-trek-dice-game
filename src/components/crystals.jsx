import React from 'react';
import { connect } from 'react-redux';

export const Crystals = ({ crystals }) =>
  <div>
    {crystals}
  </div>;

const mapStateToProps = state => ({
  crystals: state.crystals,
});

export default connect(mapStateToProps)(Crystals);
