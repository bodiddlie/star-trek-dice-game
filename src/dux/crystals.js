const DEPLETE_CRYSTALS = '[Dilithium Crystals] Deplete Crystals';
const RESET_CRYSTALS = '[Dilithium Crystals] Reset Crystals';
const ADD_CRYSTAL = '[Dilithium Crystals] Add Crystal';
const SET_CRYSTALS = '[Dilithium Crystals] Set Crystals';

export function depleteCrystals() {
  return { type: DEPLETE_CRYSTALS, payload: null };
}

export function resetCrystals() {
  return { type: RESET_CRYSTALS, payload: null };
}

export function addCrystal(amount) {
  return { type: ADD_CRYSTAL, amount };
}

export function setCrystals(amount) {
  return { type: SET_CRYSTALS, amount };
}

export const initialState = 10;

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case DEPLETE_CRYSTALS: {
      return Math.max(0, state - 1);
    }
    case RESET_CRYSTALS: {
      return 10;
    }
    case ADD_CRYSTAL: {
      return Math.min(10, state + action.amount);
    }
    case SET_CRYSTALS: {
      return action.amount;
    }
    default: {
      return state;
    }
  }
}
