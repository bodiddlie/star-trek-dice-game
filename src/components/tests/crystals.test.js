import React from 'react';
import { shallow } from 'enzyme';

import { Crystals } from '../crystals';

describe('<Crystals />', () => {
  it('should render the dilithium crystal value', () => {
    const wrapper = shallow(<Crystals crystals={4} />);
    expect(wrapper).toMatchSnapshot();
  });
});
