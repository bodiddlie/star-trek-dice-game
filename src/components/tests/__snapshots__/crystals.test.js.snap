// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Crystals /> should render the dilithium crystal value 1`] = `
<div>
  4
</div>
`;
